pkg/controller/toolchainconfig/configuration.go

```diff
@@ -0,0 +1,61 @@
+package toolchainconfig
+
+import "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+
+type ToolchainConfig struct {
+	cfg *v1alpha1.ToolchainConfigSpec
+}
+
+func (c *ToolchainConfig) AutomaticApproval() AutoApprovalConfig {
+	return AutoApprovalConfig{c.cfg.Host.AutomaticApproval}
+}
+
+func (c *ToolchainConfig) Deactivation() DeactivationConfig {
+	return DeactivationConfig{c.cfg.Host.Deactivation}
+}
+
+type AutoApprovalConfig struct {
+	approval v1alpha1.AutomaticApproval
+}
+
+func (a AutoApprovalConfig) IsEnabled() bool {
+	return getBool(a.approval.Enabled, false)
+}
+
+func (a AutoApprovalConfig) ResourceCapacityThresholdDefault() int {
+	return getInt(a.approval.ResourceCapacityThreshold.DefaultThreshold, 0)
+}
+
+func (a AutoApprovalConfig) ResourceCapacityThresholdSpecificPerMemberCluster() map[string]int {
+	return a.approval.ResourceCapacityThreshold.SpecificPerMemberCluster
+}
+
+func (a AutoApprovalConfig) MaxNumberOfUsersOverall() int {
+	return getInt(a.approval.MaxNumberOfUsers.Overall, 0)
```

Comment:
This was required too, I would set for example:
```suggestion
	return getInt(a.approval.MaxNumberOfUsers.Overall, 1000)
```